op-node/rollup/derive/isthmus_upgrade_transactions.go

```diff
@@ -7,14 +7,170 @@ import (
+
+	// Enable Isthmus Parameters
+	enableIsthmusSource = UpgradeDepositSource{Intent: "Isthmus: Gas Price Oracle Set Isthmus"}
+	enableIsthmusInput  = crypto.Keccak256([]byte("setIsthmus()"))[:4]
+
 	blockHashDeployerSource     = UpgradeDepositSource{Intent: "Isthmus: EIP-2935 Contract Deployment"}
 	blockHashDeploymentBytecode = common.FromHex("0x60538060095f395ff33373fffffffffffffffffffffffffffffffffffffffe14604657602036036042575f35600143038111604257611fff81430311604257611fff9006545f5260205ff35b5f5ffd5b5f35611fff60014303065500")
 )
 
 func IsthmusNetworkUpgradeTransactions() ([]hexutil.Bytes, error) {
+	upgradeTxns := make([]hexutil.Bytes, 0, 8)
+
+	// Deploy L1 Block transaction
+	deployL1BlockTransaction, err := types.NewTx(&types.DepositTx{
+		SourceHash:          deployIsthmusL1BlockSource.SourceHash(),
+		From:                L1BlockIsthmusDeployerAddress,
+		To:                  nil,
+		Mint:                big.NewInt(0),
+		Value:               big.NewInt(0),
+		Gas:                 425_000,
+		IsSystemTransaction: false,
+		Data:                l1BlockIsthmusDeploymentBytecode,
+	}).MarshalBinary()
+
+	if err != nil {
+		return nil, err
+	}
+
+	upgradeTxns = append(upgradeTxns, deployL1BlockTransaction)
+
+	// Deploy Gas Price Oracle transaction
+	deployGasPriceOracle, err := types.NewTx(&types.DepositTx{
+		SourceHash:          deployIsthmusGasPriceOracleSource.SourceHash(),
+		From:                GasPriceOracleIsthmusDeployerAddress,
+		To:                  nil,
+		Mint:                big.NewInt(0),
+		Value:               big.NewInt(0),
+		Gas:                 1_625_000,
```

Comment:
It's quite a corner case but this is potentially unsafe. The portal has rules to ensure that even if the maximum amount of deposit transactions are included, there is at least 1,000,000 gas left to allow room for the L1 info transaction.  But these upgrade transactions are using more gas than that and so there may not be enough gas left in the block to include all the deposit transactions. This would result in there be no possible valid block and a complete chain halt.

In the vast majority of cases, the L2 block gas limit will be far more than 1 million more than the deposit transaction gas limit and it would work, but there's nothing that guarantees that.

Reply:
We plan to fix this by introducing additional upgrade block consensus rules (upgrade gas), see https://github.com/ethereum-optimism/optimism/issues/14649. We will add this as follow up work so we can wrap up this PR.